src/legacy/ui/public/agg_types/controls/field.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { ComboBoxGroupedOptions, SubAggParamsProp } from '../../vis/editors/default';
+import { FieldParamEditor, FieldParamEditorProps } from './field';
+import { AggConfig, VisState } from '../../vis';
+
+function callComboBoxOnChange(comp: ReactWrapper, value: any = []) {
+  const comboBoxProps: EuiComboBoxProps<string> = comp.find(EuiComboBox).props();
+  if (comboBoxProps.onChange) {
+    comboBoxProps.onChange(value);
+  }
+}
+
+describe('FieldParamEditor component', () => {
+  let setValue: jest.Mock;
+  let setValidity: jest.Mock;
+  let setTouched: jest.Mock;
+  let onChange: jest.Mock;
+  let defaultProps: FieldParamEditorProps;
+  let indexedFields: ComboBoxGroupedOptions<Field>;
+  let field: Field;
+  let option: {
+    label: string;
+    target: Field;
+  };
+
+  beforeEach(() => {
+    setValue = jest.fn();
+    setValidity = jest.fn();
+    setTouched = jest.fn();
+    onChange = jest.fn();
+
+    field = { displayName: 'bytes' } as Field;
+    option = { label: 'bytes', target: field };
+    indexedFields = [
+      {
+        label: 'Field',
+        options: [option],
+      },
+    ];
+
+    defaultProps = {
+      agg: {},
+      aggParam: {
+        name: 'field',
+        type: 'field',
+        editorComponent: null as any,
+        onChange,
+      },
+      value: undefined,
+      editorConfig: {},
+      indexedFields,
+      showValidation: false,
+      setValue,
+      setValidity,
+      setTouched,
+      state: {} as VisState,
+      metricAggs: [] as AggConfig[],
+      subAggParams: {} as SubAggParamsProp,
+    };
+  });
+
+  it('should disable combo box when indexedFields is empty', () => {
+    defaultProps.indexedFields = [];
+    const comp = shallow(<FieldParamEditor {...defaultProps} />);
+
+    expect(comp.find('EuiComboBox').prop('isDisabled')).toBeTruthy();
+  });
+
+  it('should set field option value if only one available', () => {
+    mount(<FieldParamEditor {...defaultProps} />);
+
+    expect(setValue).toHaveBeenCalledWith(field);
+  });
+
+  // this is the case when field options do not have groups
+  it('should set field value if only one available', () => {
+    defaultProps.indexedFields = [option];
+    mount(<FieldParamEditor {...defaultProps} />);
+
+    expect(setValue).toHaveBeenCalledWith(field);
+  });
+
+  it('should set validity as true when value is defined', () => {
+    defaultProps.value = field;
+    mount(<FieldParamEditor {...defaultProps} />);
+
+    expect(setValidity).toHaveBeenCalledWith(true);
+  });
+
+  it('should set validity as false when value is not defined', () => {
+    mount(<FieldParamEditor {...defaultProps} />);
+
+    expect(setValidity).toHaveBeenCalledWith(false);
+  });
+
+  it('should set validity as false when there are errors', () => {
+    defaultProps.indexedFields = [];
+    mount(<FieldParamEditor {...defaultProps} />);
+
+    expect(setValidity).toHaveBeenCalledWith(false);
+
+    defaultProps.customError = 'customError';
```

Comment:
This should be a separate test. the `setValidity` mock is not reseted between the two mounts, so the second `expect(setValidity).toHaveBeenCalledWith(false);` does not validate anything.

Reply:
Done.